### PR TITLE
Instrument `mod_ping`

### PR DESCRIPTION
**Pull request by jacekwegr**
This PR adds instrumentation to `mod_ping`.

One important change is the merging of two events (`ping_response_time` and `ping_response`) into one, as they were always invoked simultaneously.

The measurement of time in `mod_ping_response` does not use `mongoose_instrument:span` because the time measurement is done through hooks.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/220484) / elasticsearch_and_cassandra_mnesia / 31117dd3619aa602cd7e1da5fa7cdb2c56636833
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4288/220484/elasticsearch_and_cassandra_mnesia.26.2.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4288/220484/elasticsearch_and_cassandra_mnesia.26.2.2-amd64/big/ct_run.test%40da35d1aeac7e.2024-05-24_13.01.38/index.html)
**OK: 457** / Failed: 0 / **User-skipped: 41** / Auto-skipped: 0

File: big_tests/tests/mod_ping_SUITE.erl [hunk 1]
```diff
@@ -155,11 +157,11 @@ disco(Config) ->
...
 ping(ConfigIn) ->
     Domain = domain(),
-    HostType = domain_helper:host_type(mim),
+    HostType = host_type(),
     HostTypePrefix = domain_helper:make_metrics_prefix(HostType),
     Metrics = [
-        {[HostTypePrefix, mod_ping, ping_response],0},
-        {[HostTypePrefix, mod_ping, ping_response_timeout],0}
+        {[HostTypePrefix, mod_ping_response, count], 0},
+        {[HostTypePrefix, mod_ping_response_timeout, count], 0}
```
Comment (chrzaszcz) [inline big_tests/tests/mod_ping_SUITE.erl:164]: The plan is to gradually remove `mongoose_metrics` - also from tests. Could you replace all such checks with assertions for events?

**Review by chrzaszcz — COMMENTED** [review]
It looks good in general, but I think we should consistently remove all `mongoose_metricd` checks form our tests. That was one of the intentions of the instrumentation rework - by removing the dependency on exometer, we would be able to switch it off in the future.

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/221725) / elasticsearch_and_cassandra_mnesia / 1c615da65eec0f0a9430c7876bbac54e6c24d9d4
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4288/221725/elasticsearch_and_cassandra_mnesia.26.2.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4288/221725/elasticsearch_and_cassandra_mnesia.26.2.2-amd64/big/ct_run.test%408af99b46d844.2024-06-10_15.21.52/index.html)
**OK: 457** / Failed: 0 / **User-skipped: 41** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/221770) / elasticsearch_and_cassandra_mnesia / 809e5584e368d8f9a750e18bdd4a99e57d403f4f
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4288/221770/elasticsearch_and_cassandra_mnesia.26.2.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4288/221770/elasticsearch_and_cassandra_mnesia.26.2.2-amd64/big/ct_run.test%40c29deda60c3a.2024-06-10_15.36.43/index.html)
**OK: 457** / Failed: 0 / **User-skipped: 41** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/221812) / elasticsearch_and_cassandra_mnesia / b9d35aa03c6d540a0e75c1f47832e8bc44d3fe13
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4288/221812/elasticsearch_and_cassandra_mnesia.26.2.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4288/221812/elasticsearch_and_cassandra_mnesia.26.2.2-amd64/big/ct_run.test%405cb8a5e924dc.2024-06-11_08.52.29/index.html)
**OK: 457** / Failed: 0 / **User-skipped: 41** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/221959) / elasticsearch_and_cassandra_mnesia / 8dc8416f108520821c8f61c9daa42144a5cf9dda
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4288/221959/elasticsearch_and_cassandra_mnesia.26.2.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4288/221959/elasticsearch_and_cassandra_mnesia.26.2.2-amd64/big/ct_run.test%40531e0f229b7c.2024-06-11_16.58.39/index.html)
**OK: 457** / Failed: 0 / **User-skipped: 41** / Auto-skipped: 0

**Review by chrzaszcz — APPROVED** [review]
Looks good 👍

**Pull request merged**